doc/adrs/130_add_more_filter_capabilities_on_diagram.adoc

```diff
@@ -0,0 +1,50 @@
+= ADR-127 - Add more filter capabilities on diagram
+
+== Context
+
+Right now, we have severals tools to filter diagram elements.
+
+=== Nodes
+
+We can hide or fade a node.
+
+In the case where the node is a compartiments, we can collapse/uncollapse. 
+
+We can also specify in the view dsl if we want to show icons label.
+
+=== Edges
+
+We can hide or fade an edge.
+
+We can also specify in the view dsl if we want to show icons label.
+
+== Decision
+
+We will add the capability to show/hide a label or an icon label on a diagram representation.
+
+Since this will bring new combinations of elements visible, hidden or partially hidden on a diagram, we will add a new representation to apply a specific view modifier to all selected elements on a diagram.
+
+This new representation will be accessible on the diagram panel and allow users to easily configure each elements view modifier
++
+.FilterDiagramElementsPanel
+image:images/130/filterElementRepresentation.png[Filter diagram elements panel,70%]
+
+The diagram element palette will also be enriched with access to new tools to show/hide label or icon labels.
+
+
+== Consequences
+
+We will add a new attribute to Node and Edges
+
+[source,java]
+----
+public enum ViewLabelModifier {
+    Visible, IconHidden, LabelHidden, Hidden
+}
+----
+
+We will also add a new representation in a dedicated package sirius-component-diagrams-filter
```

Comment:
Everything above this is related to the behavior of the feature being added, please extract that in a shape. The technical part related to how we will do it should stay in the ADR.

I have more questions related to both the behavior and technical aspect of this subject so separating them will be better to talk about each point. We will have to clarify way more things before starting with technical details like that, I'm not even sure that this `ViewLabelModifier` is relevant.

Here are some questions that need to be clarified in both a shape and an ADR.

Shape:

- What will the diagram element palette look like with the new tools?
  - Should it open something similar to the dialog in the screenshot?
- You want a representation for the dialog visible in the screenshot to keep it synchronized with the content of the diagram, right?
- This representation should not display the semantic elements but the graphical elements, right?
  - If an element is visible multiple times in the diagram, we should be able to apply different settings to each instance, right?
  - You want to display the structure of the diagram as a tree, so this representation would take the current diagram as an input, we may be talking about a representation using another representation as an input just like in a portal.

ADR:

- You are talking about a new representation, were you thinking about something from scratch or a form based one? Since forms can already contain textfields, trees and buttons, we may not be that far from what you are requiring.
  - Let's consider that we would use a form based one, what kind of capabilities are missing in such a form (checkbox in the tree for one, filter is another, drop down button is a third one, icons after the label in the tree is another one)
  - You would also need something like the long requested feature (prototyped by @lfasani) to have some sort of configurable state global the form to have one field interact with another one (the filter can change the content of the tree)
  - If you are not considering a form based representation, why?

Reply:
Thanks for the in depth review, I will rewrite the ADR and add a Shape.